File: frontend/lib/Classes/contributionModel.dart
import 'package:frontend/Classes/Event.dart';
import 'package:frontend/Classes/registrationModel.dart';
import 'package:json_annotation/json_annotation.dart';
part 'contributionModel.g.dart';

@JsonSerializable()
class ContributionModel {
  ContributionModel(
      this.eventID,
      this.contributorID,
      this.points,
      this.registrationDetails,
      this.eventName,
      this.eventDescription,
      this.eventLogo,
      this.type);

  factory ContributionModel.fromJson(Map<String, dynamic> data) =>
      _$ContributionModelFromJson(data);
  Map<String, dynamic> toJson() => _$ContributionModelToJson(this);

  final String eventName;
  final String eventDescription;
  final String eventLogo;
  final String type;
  final String eventID;
  final String contributorID;
  final int points;
  final String registrationDetails;
}

File: Backend/models/Events/contributionModel.js
const mongoose = require('mongoose');

const contributionModelSchema = new mongoose.Schema({
    eventName : String,
    eventDescription : String,
    eventLogo : String,
    type : String,
    eventID : {type : mongoose.Schema.Types.ObjectId, ref : "roboEvent"},
    contributorID : {type : mongoose.Schema.Types.ObjectId, ref : "User"},
    points : Number,
    registrationDetails : {type : mongoose.Schema.Types.ObjectId, ref : "registrationModel"},
}, {collection : "Contributions"});

const contributionModel = mongoose.model('contributionModel', contributionModelSchema);
module.exports = contributionModel;